### DISCUSSION TITLE
Lora walkie Talkie

### DISCUSSION THREAD

**Discussion by GQster** [ropg/heltec_esp32_lora_v3#31] [Q&A]
I recently had the idea to make a little comms system for fun.
I have two of the 'ordinary'  esp32 lora V3 boards and would like to know if this is doable with this hardware. 
The idea is to use lora as a walkie-talkie system. I wouldn't want a push-to-talk button though. 

A little more detail; I'm thinking of connecting this to my computer/phone and having that act as the mic/speaker. So the esp32 would only send the data via lora (no mic/speaker on the esp32 itself). 

I have yet to look into it, but would loraWAN work for this too? ie, if there are say 10 of these systems strung along so each one only ever 'sees' the one to the left and right of it, would it be able to relay the info down the chain? 

Im thinking there will be a bandwidth issue for sending voice.

Has anyone attempted something like this?

**Comment by ropg** [comment]
LoRa provides bitrates between 290 bps and 50 kbps. Voice codecs go down to ~5 kbps or so for really crummy sound quality. Given that regulations do not allow you to transmit more than a really small percentage of the time, LoRa probably doesn't provide the kind of bandwidth you want.  Also the board's radio is half-duplex, meaning you would have a push-to-talk switch, or have it be voice-activated in some way.

As for relaying: every station along the way introduces delay and a probability of errors that would need retransmission, creating the need for some kind of of protocol to handle that.

If you haven't, maybe have a spin trying out Meshtastic (which is text only) to get a feel for the real-world possibilities and limitations.

**Comment by GQster** [comment]
ok thanks!

**Comment by alex-bristol** [comment]
You might find this documentation helpful, it explains best practice and when and when not to use LoRa, it states voice data and real time data are not suitable:

https://www.thethingsnetwork.org/docs/lorawan/limitations/
